SiteWide ACL: Milestone 2.3

1995YogeshSharma opened the pull request:
This PR implements decorators for collection editors, moderators and admins.

Note: backend tests will not pass as some of the method used (like exploration_rights.is_owner) are not in this PR

1995YogeshSharma committed:
test and decorator added can_edit_collection.

1995YogeshSharma committed:
added tests and decorator can_manage_email_dashboard.

1995YogeshSharma committed:
updated test.

1995YogeshSharma committed:
tests and decorators added for moderator controller.

1995YogeshSharma committed:
added tests and decorators for profile and preference update.

1995YogeshSharma committed:
shifted require_super_admin to acl_decorators as new decorator.

1995YogeshSharma committed:
test updated.

1995YogeshSharma committed:
decorator added for uploading exploration.

1995YogeshSharma committed:
lint fixes.

seanlip commented:
Hi @1995YogeshSharma -- just so I understand, what is your merge strategy? Are you planning to merge the 2.2 PR and then merge this one? I guess I'm a bit confused because the 2.2 PR has comments that haven't been addressed yet.

If it's helpful I'm fine with you consolidating both of these into one PR that works (passes tests, is ready for merge, etc). Let us know what you want to do.

seanlip commented:
Also FYI on a quick glance I don't see any issues in this PR. It looks good!

1995YogeshSharma commented:
Hi @seanlip ....

I have partially addressed the comments on PR milestone-2.2. I thought I would first complete the work for this PR as well so that I get enough time to get it reviewed properly.

We can do it both ways. Let me handle the reviews on milestone-2.2 then we can maybe make both of them into one. I am not sure.
(Also, I had a few doubts regarding this one which I would ask today. Got too sleepy yesterday :stuck_out_tongue: )

seanlip commented:
OK, whatever works best! Let us know if you need our input on anything.

On Wednesday, July 19, 2017, Yogesh Sharma <notifications@github.com> wrote:

> Hi @seanlip <https://github.com/seanlip> ....
>
> I have partially addressed the comments on PR milestone-2.2. I thought I
> would first complete the work for this PR as well so that I get enough time
> to get it reviewed properly.
>
> We can do it both ways. Let me handle the reviews on milestone-2.2 then we
> can maybe make both of them into one. I am not sure.
> (Also, I had a few doubts regarding this one which I would ask today. Got
> too sleepy yesterday 😛 )
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/oppia/oppia/pull/3661#issuecomment-316575252>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AKFeym2qA9t5fUTsd--NJkmntrTtfVM6ks5sPrZAgaJpZM4OdHf4>
> .
>

seanlip commented:
Hi @1995YogeshSharma -- the other PR is merged! Could you please fix the conflicts, then let's get this in?

1995YogeshSharma committed:
Merge remote-tracking branch 'upstream/develop' into milestone-2.3

Conflicts:
	core/domain/acl_decorators.py
	core/domain/acl_decorators_test.py
	core/domain/role_services.py

1995YogeshSharma committed:
added decorator for CollectionRightsHandler

1995YogeshSharma commented:
done.

seanlip reviewed (approved):
LGTM. Thanks!

1995YogeshSharma committed:
fixed failing tests. 1.) added function to set email senders in test utils (it was directly set currently which led to authorization failure) 2.) Removed decorators from get methods where open access was needed.

1995YogeshSharma committed:
lint fixes.

seanlip commented:
Hang on. For GET methods open to guests, don't we still need a decorator (if we're going to check later that every handler method has a decorator)?

1995YogeshSharma commented:
Yep... We had two options for handling such cases. 
1.) Create a handler whitelist to skip test for them.
2.) Create a decorator that allows open access.

I am going to introduce a new decorator that will just attach the __wrapped__ to the function for check and nothing else.
This will be done in PR for that test (checking decorator for every handler).

seanlip commented:
OK, sounds good. Thanks!

1995YogeshSharma committed:
fix for failing backend test.
